components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/executers/RemoteFetchTaskExecutor.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.remotefetch.core.executers;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.remotefetch.common.RemoteFetchConfiguration;
+import org.wso2.carbon.identity.remotefetch.common.RemoteFetchCoreConfiguration;
+import org.wso2.carbon.identity.remotefetch.core.executers.tasks.RemoteFetchConfigurationBatchTask;
+import org.wso2.carbon.identity.remotefetch.core.executers.tasks.RemoteFetchConfigurationImmediateTask;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This class is a execution engine for trigger actions.
+ * This execution engine has functions to start batch task and immediate task.
+ * This execution engine starts batch task at component activation and schedule at fixed rate.
+ * {@link RemoteFetchConfigurationBatchTask}
+ * This execution engine starts immediate task when OSGi trigger service called.
+ * {@link RemoteFetchConfigurationImmediateTask}
+ *
+ */
+public class RemoteFetchTaskExecutor {
+
+    private static final Log log = LogFactory.getLog(RemoteFetchTaskExecutor.class);
+
+    private static RemoteFetchTaskExecutor instance = new RemoteFetchTaskExecutor();
+    private ScheduledExecutorService scheduler;
+    RemoteFetchConfigurationBatchTask remoteFetchConfigurationBatchTask;
+
+    public RemoteFetchTaskExecutor() {
+
+    }
+
+    public static RemoteFetchTaskExecutor getInstance() {
+
+        return instance;
+    }
+
+    /**
+     * Create new thread pool for scheduler.
+     */
+    public void createScheduler() {
+
+        scheduler = Executors.newScheduledThreadPool(1);
+    }
+
+    /**
+     * Shutdown scheduler while deactivating the component.
+     */
+    public void shutdownScheduler() {
+
+        scheduler.shutdown();
+    }
+
+    /**
+     * Schedule Batch execution after checking whether enabled or not.
+     * @param remoteFetchCoreConfiguration
+     */
+    public void startBatchTaskExecution(RemoteFetchCoreConfiguration remoteFetchCoreConfiguration) {
+
+        if (remoteFetchCoreConfiguration.isEnableCore()) {
```

Comment:
This if logic needs to go to the place where this method is called. this method needs to change to 
public void startBatchTasksExecution() {